02-typescript/02-todo-list/src/main.ts

```diff
@@ -28,33 +31,62 @@ const toDos: State = [
   },
 ];
 
+
 const crtToDo = () => {
   getDiv.innerHTML= "";
-  let crtInput = document.createElement("input");
+ 
   crtInput.setAttribute("type","text");
   crtInput.innerHTML= " ";
   getDiv.appendChild(crtInput);
-  let crtUl = document.createElement("ul");
-  getDiv.appendChild(crtUl)
+  
+  getDiv.appendChild(crtUl);
+  
+  let crtButton = document.createElement('button');
+  listTodo();
+  crtButton.innerText ="Add";
+  getDiv.appendChild(crtButton)
+  crtButton.addEventListener("click",addTodo);
+  crtFilter?.addEventListener("click",function(){
+    listTodo(crtFilter?.checked);
+  });
+}
+
 
-  for(const todo of toDos) {
+const addTodo =()=>{
+  toDos.push({
+    'title':crtInput.value,
+    'done':false
+  });
+  listTodo();
+}
+
+const listTodo=(filter?:boolean)=>{
```

Comment:
Here the question mark means that the parameter is optional. You needed it because you call `listTodo()` without argument at line 60. I would prefer having a mandatory argument, and calling `listTodo(false)` when there is no filter (by the way, note that this will disable the filter after adding a to-do).